kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java

```diff
@@ -228,6 +228,10 @@ public static String getNamespace(HasMetadata entity) {
    */
   public static String sanitizeName(String name) {
     if(name != null) {
+      final int length = name.length();
+      if(length > KUBERNETES_DNS1123_LABEL_MAX_LENGTH) {
+        name = name.substring(0, KUBERNETES_DNS1123_LABEL_MAX_LENGTH);
+      }
```

Comment:
Makes sense. My take was that I would truncate first so that we wouldn't need to replace characters that we know won't be in the final name anyway… Plus I'm not sure that the replacement takes care of repetitions so I'm not sure I actually understand your example… 😅

Reply:
![image](https://user-images.githubusercontent.com/488391/134200220-01a11afe-b03f-4189-82ff-e70c1ac4c91f.png)

![image](https://user-images.githubusercontent.com/488391/134200117-d97a2f82-f90a-4952-be3f-3870b619b704.png)

This is more of a mixed corner and edge case situation. However, since this is regarding resource names, it's probably better to preserve the greatest amount of characters to avoid name collisions.

Reply:
True.

Reply:
How about changing the regex to:

```
public static final Pattern KUBERNETES_DNS1123_LABEL_REGEX = Pattern.compile("[a-z0-9]([-a-z0-9]{0,61}[a-z0-9])?");
```

And consider a method like:
```
  public static String sanitizeName(String name) {
    if(name != null) {
      name = name.replaceAll("[^-A-Za-z0-9]+", "-");
      if (name.length() > 63) {
        name = name.substring(0, 63);
      }
      name = name.toLowerCase();
      if (!Character.isLetterOrDigit(name.charAt(0))) {
        name = "a" + name;
      }
      if (!Character.isLetterOrDigit(name.charAt(name.length() - 1))) {
        name = name.substring(0, Math.min(62, name.length())) + "a";
      }
    }
    return name;
  }
```